ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandRunner.java

```diff
@@ -346,14 +354,77 @@ public CommandQueue getCommandQueue() {
     return commandStore;
   }
 
+  public CommandRunnerStatus checkCommandRunnerStatus() {
+    if (state.getStatus() == CommandRunnerStatus.DEGRADED) {
+      return CommandRunnerStatus.DEGRADED;
+    }
+
+    final Pair<QueuedCommand, Instant> currentCommand = currentCommandRef.get();
+    if (currentCommand == null) {
+      state = lastPollTime.get() == null
+          || Duration.between(lastPollTime.get(), clock.instant()).toMillis()
+              < NEW_CMDS_TIMEOUT.toMillis() * 3
+              ? new Status(CommandRunnerStatus.RUNNING, CommandRunnerDegradedReason.NONE)
+                  : new Status(CommandRunnerStatus.ERROR, CommandRunnerDegradedReason.NONE);
+      
+    } else {
+      state = Duration.between(currentCommand.right, clock.instant()).toMillis()
+        < commandRunnerHealthTimeout.toMillis()
+        ? new Status(CommandRunnerStatus.RUNNING, CommandRunnerDegradedReason.NONE)
+              : new Status(CommandRunnerStatus.ERROR, CommandRunnerDegradedReason.NONE);
+    }
+
+    return state.getStatus();
+  }
+
+  public CommandRunnerDegradedReason getCommandRunnerDegradedReason() {
```

Comment:
I'd drop these methods and just have 1 get CommandRunnerStatus() method that callers can call to get the current status, and then the caller can pull out the info that is of interest.